features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/AsnInfo.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.protocols.bmp.adapter;
+
+import java.util.Optional;
+
+public class AsnInfo {
+
+    private Long asn;
+    private String asName;
+    private String orgId;
+    private String orgName;
+    private String address;
+    private String city;
+    private String stateProv;
+    private String postalCode;
+    private String country;
+    private String source;
+    private String rawOutput;
+    private String remarks;
+
+    public AsnInfo(Long asn, String source, String rawOutput) {
+        this.asn = asn;
+        this.source = source;
+        this.rawOutput = rawOutput;
+    }
+
+    public Long getAsn() {
+        return asn;
+    }
+
+    public void setAsn(Long asn) {
+        this.asn = asn;
+    }
+
+    public String getAsName() {
+        return asName;
+    }
+
+    public void setAsName(String asName) {
+        this.asName = asName;
+    }
+
+    public String getOrgId() {
+        return orgId;
+    }
+
+    public void setOrgId(String orgId) {
+        this.orgId = orgId;
+    }
+
+    public String getOrgName() {
+        return orgName;
+    }
+
+    public void setOrgName(String orgName) {
+        this.orgName = orgName;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public String getCity() {
+        return city;
+    }
+
+    public void setCity(String city) {
+        this.city = city;
+    }
+
+    public String getStateProv() {
+        return stateProv;
+    }
+
+    public void setStateProv(String stateProv) {
+        this.stateProv = stateProv;
+    }
+
+    public String getPostalCode() {
+        return postalCode;
+    }
+
+    public void setPostalCode(String postalCode) {
+        this.postalCode = postalCode;
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    public void setCountry(String country) {
+        this.country = country;
+    }
+
+    public String getSource() {
+        return source;
+    }
+
+    public void setSource(String source) {
+        this.source = source;
+    }
+
+    public String getRawOutput() {
+        return rawOutput;
+    }
+
+    public void setRawOutput(String rawOutput) {
+        this.rawOutput = rawOutput;
+    }
+
+    public String getRemarks() {
+        return remarks;
+    }
+
+    public void setRemarks(String remarks) {
+        this.remarks = remarks;
+    }
+
+    public static AsnInfo parseOutput(Long asn, String source, String rawOutput) {
+        AsnInfo asnInfo = new AsnInfo(asn, source, rawOutput);
+        // Split output into lines
+        String[] lines = rawOutput.split("\\r?\\n");
+        for (String line : lines) {
```

Comment:
As far as I know, there can be entries can appear more than once. For instance there can be multiple address entries if the address is more than one line. The OpenBMP-implementation concatenates these strings before persisting them. So, this is missing here.

Reply:
yeah, OpenBMP has this handling for RIPE. I m looking for an example AS number to properly implement this.  Currently all the responses I'm getting are from ARIN and RTT.

Reply:
I updated few things but still didn't get an example for having Address in multiple lines even with RIPE. We can address this later if we find such scenarios.

Reply:
Ok

Reply:
@cgorantla https://www.ripe.net/manage-ips-and-asns/db/support/querying-the-ripe-database has some examples showing multiple address lines. What do you need for testing? Maybe I can provide something.

Reply:
I was looking for an AS number that can give us output just like above example.  
We need to split multiple address lines into  address, city,  state and country. I'm looking if there is any standard way to do that.

Reply:
My understanding of the code above is that it will overwrite with the last encountered address value. Maybe it would be more suitable to concatenate the multiple lines separated by `\n`. Looks like that's what openbmp is doing: https://github.com/SNAS/openbmp-mysql-consumer/blob/master/cron_scripts/gen-whois/gen-whois-asn.py#L186 .

The reference code calls the `whois` command, which by default also fetches related entries. Maybe `org.apache.commons.net.whois.WhoisClient` does not do this?

`whois -h whois.ripe.net AS8319` shows some output with multiple address lines.

This is not only true for the `address` field but for almost all fields (RIPE marks them as `multiple` in the link above).

Reply:
Removing optional a with the query `a AS8319` seems to give the output identical to `whois` client.  Updated code to append multiple lines in address/remarks/orgId as per link above.

Reply:
@fooker  This is taken care with dc3c5b6.  Can you take a look